### ISSUE TITLE
Difficult to find broker and federate configuration string options

### ISSUE THREAD

**Issue by bpalmintier** [GMLC-TDC/HELICS#1823]
It is hard to find information about non-default options to pass to the broker and federate init strings.

I ran into this option trying to expand the python pisender/pireceiver example to work between two separate computers. Doing so requires specifying the IP address for the broker and configuring the broker to listen to that ip address (rather than just localhost). For this specific problem I eventually talked to someone to find I needed: `--broker_address=XXX` in the federate init string and "--interface=XXX" in the broker init string, but I could not find the information with a good bit of hunting around the documentation.

Suggestions:
1. It would be good to add a simple multi-machine example to the User Guide (@trevorhardy ?)
2. More generally, though it would be very helpful to expose commonly changed default settings in our examples. For instance if in pisender.py we changed the init string lines to
```
brokerinitstring = "-f 2 --name=mainbroker --log_level=7 --dumplog --interface=127.0.0.1 --port=23405"
federateinitstring = "--broker=mainbroker --log_level=7 --federates=1 --dumplog --broker_address=127.0.0.1  --port=23405"
```
then even without documentation it is easy for most coders to know what to change. Alternatively, we could provide the full set of defaults as a comment above each line. If this type of code or comment existed throughout our examples and documentation it would hopefully be much easier for users to adapt for their needs.

Thoughts?

**Comment by trevorhardy** [comment]
Examples and documentation for this are in the v3 User Guide; at this time I'm not aware of any plans to update the v2 documentation to cover this.

[User Guide](https://docs.helics.org/en/helics3userguide/user-guide/advanced_topics/broker_hierarchies.html)

[Partly relevant example](https://github.com/GMLC-TDC/HELICS-Examples/tree/master/user_guide_examples/advanced/advanced_brokers/simultaneous)

[Configuration options reference](https://docs.helics.org/en/helics3userguide/references/configuration_options_reference.html#general-federate-options)

**Comment by bpalmintier** [comment]
Great to see these new parts to the forthcoming User Guide, but I still think they are hard to find:

 - [ ] Notably the User Guide link is for multiple brokers. Seems we need a simpler and easier to find example titled something like "Running HELICS across multiple machines"
 - [ ] Also the configuration options reference, in addition to being quite long and hard to parse for such a simple example, seems to only cover federate configuration options, not broker options
 - [ ] And I still think we need to add commonly changed defaults to our example files

**Comment by trevorhardy** [comment]
1. We could add a section to the User Guide about running across multiple machines. I'll have to think about the structure of the Guide to figure out where it goes.
2. The configuration options reference is long because, as you can see, there are lots of options. The "General Options" does show the `broker_address` as part of the `core_init_string` discussion; this is how a federate is configured.
3. When you say "commonly changed defaults" I take this to mean you want to add several/many configuration options to all the example files and set their values to the default options; correct? I'm open to such an idea with two hesitations:
 - What do you proposed as the list of commonly changed options and how do we figure out which options they are?
 - Adding these makes it not obvious that they are optional (that is, they are being set to their defaults) and will obfuscate what actually **MUST** be included for a particular use case.

**Comment by bpalmintier** [comment]
1. Sounds great
2. Yeah lots there. But I couldn't find a similar for the broker config string passed to `broker = h.helicsCreateBroker("zmq", "", brokerinitstring)` In many cases you have to add the `--interface=$IPADDRESS` option to this broker config string for it to accept external connections. I'm sure there is also a slew of other options for this string to, but don't know of a similar reference to find them all.
3. Good point about optional vs. required. Nadia suggested that maybe we put a comment line above with either additional options or a duplicate with more settings revealed. Proposed set of options to start with
   * broker IP address
   * broker port
   * something around broker timeout
   * (open to others)

**Comment by trevorhardy** [comment]
2. Yeah, there are a ton of options for the `x_init_string` and many of them are duplicated for both the broker and the core. We need to ask @phlptp but I VERY surprised that the `interface` option was required for you when working on multiple machines. We've never had to add the `interface` option on our multi-machine HELICS co-sims.
3. Comments aren't supported in JSON so we'd either need to add support for them in our code base or require users to delete out all the comment lines before using the JSON.

**Comment by trevorhardy** [comment]
3. Other suggested defaults:

- `log_level`
- `terminate_on_error`
- `autobroker`
- `only_update_on_change`
- `uniterruptible`
- `period`
- `offset`
- `time_delta`
- `wait_for_current_time_update`
- (value) `connection_optional`
- (value) `only_transmit_on_change` and `only_update_on_change`
- (value) `tolerance`
- (message) `target`

**Comment by bpalmintier** [comment]
For 3. I was thinking of the non-json config examples, such as the python and other language examples

iPhone iTypo iSorry
________________________________

**Comment by trevorhardy** [comment]
[The v3 User Guide has a page in the Fundamental Topics section on federate configuration](https://docs.helics.org/en/latest/user-guide/fundamental_topics/interface_configuration.html) and I've added a few more options including `broker_address` in `federate_init_string`. 

All of the examples to date are designed to run on a single machine. We could add a special case that would require two machines. Let me see if I can set up some VMs so that I can test whatever example I might be able to put together.

**Comment by trevorhardy** [comment]
PR #2237 has added a User Guide chapter on setting up a multi-computer co-simulation and a corresponding example. I'm going to close this out for now unless there are further concerns.
